interfaces/cython/cantera/mixture.pyi

```diff
@@ -0,0 +1,65 @@
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+from collections.abc import Sequence
+from typing import Any, Literal
+
+from ._types import Array, ArrayLike, EquilibriumSolver, PropertyPair
+from .thermo import ThermoPhase
+
+class Mixture:
+    def __init__(self, phases: Sequence[tuple[ThermoPhase, float]]) -> None: ...
+    def report(self, threshold: float = 1e-14) -> str: ...
+    def __call__(self, *args: Any, **kwargs: Any) -> None: ...
+    @property
+    def n_elements(self) -> int: ...
+    def element_index(self, element: str | bytes | float) -> int: ...
+    @property
+    def n_species(self) -> int: ...
+    def species_name(self, k: int) -> str: ...
+    @property
+    def species_names(self) -> list[str]: ...
+    def species_index(
+        self, phase: ThermoPhase | str | int, species: str | bytes | int
+    ) -> int: ...
+    def n_atoms(self, k: int, m: int) -> int: ...
+    @property
+    def n_phases(self) -> int: ...
+    def phase(self, n: int) -> ThermoPhase: ...
+    def phase_index(self, p: str) -> int: ...
+    @property
+    def phase_names(self) -> list[str]: ...
+    @property
+    def T(self) -> float: ...
+    @T.setter
+    def T(self, T: float) -> None: ...
+    @property
+    def min_temp(self) -> float: ...
+    @property
+    def max_temp(self) -> float: ...
+    @property
+    def P(self) -> float: ...
+    @P.setter
+    def P(self, P: float) -> None: ...
+    @property
+    def charge(self) -> float: ...
+    def phase_charge(self, p: str) -> float: ...
```

Comment:
`p` can also be an `int` or a `ThermoPhase` here, and several other methods that pass `p` through the `phase_index` method.